boards: arm: efm32gg_*: Enable watchdog, SWO and 72Mhz operation

thorvald opened the pull request:
This makes the EFM32GG11-based development kits use the 72Mhz HF RC-Oscillator for the CPU, rather than the 50Mhz HFXO crystal. The HFXO is only enabled if needed to drive the ethernet clock.
This also enables the watchdog, the TRNG and SWO output, and adds support for OpenOCD (but leaves the default as jlink).

Signed-off-by: Thorvald Natvig <thorvald@natvig.com>

chrta commented:
@thorvald I have some issues testing this on my efm32gg_stk3701a. It seems something is broken with the ethernet driver (also on the zephyr master branch).I cannot get a lease with samples/net/dhcpv4_client. Does this work for you? It works on the nucleo_f429zi board.

chrta commented on dts/arm/silabs/efm32gg11b.dtsi:
```diff
@@ -288,6 +289,33 @@
+			trng0: trng@4001d000 {
+				compatible = "silabs,gecko-trng";
+				reg = <0x4001d000 0x400>;
+				interrupts = <49 0>;
```
49 -> 66 see https://www.silabs.com/documents/public/reference-manuals/efm32gg11-rm.pdf

thorvald replied:
D'oh! Fixed.

chrta reviewed (changes requested):
Otherwise looks good to me. I successfully tested openocd and ethernet on efm32gg_stk3701a

MaureenHelm reviewed (changes requested):
Please split the changes into 4 separate commits: watchdog, clocks, TRNG, and openocd. It's ok to keep in the same PR.

thorvald committed:
boards: arm: efm32gg_*: Enable watchdog

This defines wdog0 and wdog1 for the EFM32GG11 MCU and enables wdog0
in the dts for the matching development kits.

Signed-off-by: Thorvald Natvig <thorvald@natvig.com>

thorvald committed:
boards: arm: efm32gg_*: Enable TRNG

This defines the TRNG for EFM32GG11 and enables it for trng0
in the dts for the matching development kits.

Signed-off-by: Thorvald Natvig <thorvald@natvig.com>

thorvald committed:
boards: arm: efm32gg_*: Use 72Mhz HFRCO to drive CPU

This makes the 72Mhz HFRCO drive the MCU HF clock, and only uses
the HFXO to drive the ethernet clock.

Signed-off-by: Thorvald Natvig <thorvald@natvig.com>

thorvald committed:
boards: arm: efm32gg_*: Enable SWO logging

This enables SWO logging for the EFM32GG11 development kits.

Signed-off-by: Thorvald Natvig <thorvald@natvig.com>

thorvald committed:
boards: arm: efm32gg_*: Enable OpenOCD

This enables OpenOCD for the EFM32GG11 development kits, but keeps
the default as jlink.

Signed-off-by: Thorvald Natvig <thorvald@natvig.com>

thorvald commented:
Split into independent commits.

chrta reviewed (approved):
LGTM
